pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -94,17 +95,49 @@ func (pr *PipelineRun) IsGracefullyStopped() bool {
 
 // GetTimeout returns the the applicable timeout for the PipelineRun
 func (pr *PipelineRun) GetTimeout(ctx context.Context) time.Duration {
-	// Use the platform default if no timeout is set
-	if pr.Spec.Timeout == nil && pr.Spec.Timeouts == nil {
-		defaultTimeout := time.Duration(config.FromContextOrDefaults(ctx).Defaults.DefaultTimeoutMinutes)
-		return defaultTimeout * time.Minute
-	}
-
 	if pr.Spec.Timeout != nil {
 		return pr.Spec.Timeout.Duration
 	}
+	if pr.Spec.Timeouts != nil && pr.Spec.Timeouts.Pipeline != nil {
+		return pr.Spec.Timeouts.Pipeline.Duration
+	}
+	return time.Duration(config.FromContextOrDefaults(ctx).Defaults.DefaultTimeoutMinutes) * time.Minute
+}
+
+// TasksTimeout returns the the tasks timeout for the PipelineRun, if set,
+// or the tasks timeout computed from the Pipeline and Finally timeouts, if those are set.
+func (pr *PipelineRun) TasksTimeout() *metav1.Duration {
+	if pr.Spec.Timeouts == nil {
+		return nil
+	}
+	if pr.Spec.Timeouts.Tasks != nil {
+		return pr.Spec.Timeouts.Tasks
+	}
+	if pr.Spec.Timeouts.Pipeline != nil && pr.Spec.Timeouts.Finally != nil {
+		if pr.Spec.Timeouts.Pipeline.Duration == apisconfig.NoTimeoutDuration || pr.Spec.Timeouts.Finally.Duration == apisconfig.NoTimeoutDuration {
+			return nil
+		}
+		return &metav1.Duration{Duration: (pr.Spec.Timeouts.Pipeline.Duration - pr.Spec.Timeouts.Finally.Duration)}
+	}
+	return nil
```

Comment:
optional bikeshed: since `pr.Spec.Timeouts` is very repetitive, would breaking this out into a var help readability?
```suggestion
	t := pr.Spec.Timeouts
	if t == nil {
		return nil
	}
	if t.Tasks != nil {
		return t.Tasks
	}
	if t.Pipeline != nil && t.Finally != nil {
		if t.Pipeline.Duration == apisconfig.NoTimeoutDuration || t.Finally.Duration == apisconfig.NoTimeoutDuration {
			return nil
		}
		return &metav1.Duration{Duration: (t.Pipeline.Duration - t.Finally.Duration)}
	}
	return nil
```